test/framework/src/main/java/org/elasticsearch/geo/GeometryTestUtils.java

```diff
@@ -97,6 +97,9 @@ public static Point randomPoint(boolean hasAlt) {
 
     public static Polygon randomPolygon(boolean hasAlt) {
         org.apache.lucene.geo.Polygon lucenePolygon = GeoTestUtil.nextPolygon();
+        while(area(lucenePolygon) == 0) {
```

Comment:
We often use `randomValueOtherThanMany(p -> area(p) == 0, GeoTestUtil::nextPolygon);` to do this kind of thing. It isn't faster or better than what you have, but it is more common.